xarray/tests/test_dataset.py

```diff
@@ -3495,6 +3495,25 @@ def test_setitem_str_dtype(self, dtype):
 
         assert np.issubdtype(ds.x.dtype, dtype)
 
+    def test_setitem_using_list(self):
+
+        # assign a list of variables
+        var1 = Variable(["dim1"], np.random.randn(8))
+        var2 = Variable(["dim1"], np.random.randn(8))
+        data1 = create_test_data()
+        data1[["A", "B"]] = [var1, var2]
+        data2 = data1.copy()
+        data2[["A", "B"]] = [var1, var2]
+        assert_identical(data1, data2)
```

Comment:
I would have a test that adds variables with the existing approach — one at a time — and then compares that result to adding them in the proposed approach — with a list of keys.

Reply:
Ok this should look better now :)

Reply:
Perfect